### PR TITLE
Migrate `tests/store/tracking/test_file_store.py` to pytest

### PR DESCRIPTION
**Pull request by viditjain99**
## Related Issues/PRs

 #7201 

Migrate `tests/store/tracking/test_file_store.py` to pytest
Added helper class in `test_file_store.py` for setup and tear-down functions.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/tracking/test_file_store.py [hunk 1]
```diff
@@ -315,13 +316,22 @@ def test_search_experiments_pagination(store):
     assert experiments.token is None
 
 
-class TestFileStore(unittest.TestCase, AbstractStoreTest):
+class TestFileStoreHelper:
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:319]: Let's remove this class.

File: tests/store/tracking/test_file_store.py [hunk 2]
```diff
@@ -422,1279 +432,1388 @@ def _create_root(self, root):
...
-    def test_get_experiment(self):
-        fs = FileStore(self.test_root)
-        for exp_id in self.experiments:
-            self._verify_experiment(fs, exp_id)
+@pytest.fixture
+def helper():
+    helper_obj = TestFileStoreHelper()
+    helper_obj.setUp()
+    yield helper_obj
+    helper_obj.tearDown()
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:441]: We can use the `store` fixture. Let's remove this.

File: tests/store/tracking/test_file_store.py (hunk 1 above)
Comment (viditjain99) [reply tests/store/tracking/test_file_store.py:319]: Added this class as there were a lot of data variables being created in the `_create_root()` function. Can you suggest a better way to manage these variables without creating a class?

Comment (harupy) [reply tests/store/tracking/test_file_store.py:319]: If so, let's remove `_create_root` and refactor each test.

Comment (viditjain99) [reply tests/store/tracking/test_file_store.py:319]: Yes, we can do that but the only drawback I see is that there will be a lot of redundant operations involved in recreating runs/experiments for every test.

Comment (harupy) [reply tests/store/tracking/test_file_store.py:319]: I think that's fine. Creating required resources in a test is easier to understand and veriry than creating required resources in a fixture.

```
def test_search_runs(fixture):
    runs = search_runs()  # where do we create runs...?
    assert len(runs) == 3
```

```
def test_search_runs(fixture):
    create_run("a")
    create_run("b")
    runs = search_runs()
    ...
```

Comment (viditjain99) [reply tests/store/tracking/test_file_store.py:319]: Understood! I'll make the required changes.

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by Luuut** [comment]
tests/store/artifact/test_databricks_artifact_repo.py

**Pull request merged**